source/fuzz/fuzzer.h

```diff
@@ -71,8 +72,64 @@ class Fuzzer {
       protobufs::TransformationSequence* transformation_sequence_out) const;
 
  private:
-  struct Impl;                  // Opaque struct for holding internal data.
-  std::unique_ptr<Impl> impl_;  // Unique pointer to internal data.
+  // A *pass recommendation* is a fuzzer pass together with an integer *age*
+  // indicating how long ago this fuzzer pass was recommended.  Each time some
+  // fuzzer pass is applied, the age of all recommended passes increases by
+  // one.  A recommended pass gets removed either because it has been acted on
+  // (i.e., the pass has been applied), or because the recommendation gets too
+  // old (i.e., the age component reaches some limit).
+  using RecommendedPasses = std::vector<std::pair<FuzzerPass*, uint32_t>>;
```

Comment:
Just an observation (not really related to the overall design): I would rather see a vector of X (where X could be PassWithAge or something like that). I feel like "recommended passes" works better as just the variable name rather than the type.

Reply:
I've done this a different way now.